client/src/components/modal.tsx

```diff
@@ -0,0 +1,162 @@
+import axios from 'axios';
+import React, { ChangeEvent, FormEvent, useRef, useState } from 'react';
+import { IAppointment } from '../../../shared/interfaces/appointment.interface';
+
+const Modal: React.FC = () => {
+    const [showModal, setShowModal] = useState(false);
+    const [error, setError] = useState('');
+    const nameField = useRef<HTMLInputElement>(null);
```

Comment:
Any current use for these useRefs? Not sure if you have an intended future use.

Reply:
i just wanted my input fields to be mutable so I just used useRef just in case